code/game/objects/structures/loom.dm

```diff
@@ -10,32 +8,25 @@
 	density = TRUE
 	anchored = TRUE
 
-/obj/structure/loom/attackby(obj/item/I, mob/user)
-	if(weave(I, user))
-		return
-	return ..()
-
 /obj/structure/loom/wrench_act(mob/living/user, obj/item/I)
 	..()
 	default_unfasten_wrench(user, I, 5)
 	return TRUE
 
-///Handles the weaving.
-/obj/structure/loom/proc/weave(obj/item/stack/sheet/cotton/W, mob/user)
-	if(!istype(W))
-		return FALSE
-	if(!anchored)
-		user.show_message("<span class='notice'>The loom needs to be wrenched down.</span>", MSG_VISUAL)
-		return FALSE
-	if(W.amount < FABRIC_PER_SHEET)
-		user.show_message("<span class='notice'>You need at least [FABRIC_PER_SHEET] units of fabric before using this.</span>", MSG_VISUAL)
-		return FALSE
-	user.show_message("<span class='notice'>You start weaving \the [W.name] through the loom..</span>", MSG_VISUAL)
-	if(W.use_tool(src, user, W.pull_effort))
-		if(W.amount >= FABRIC_PER_SHEET)
-			new W.loom_result(drop_location())
-			W.use(FABRIC_PER_SHEET)
-			user.show_message("<span class='notice'>You weave \the [W.name] into a workable fabric.</span>", MSG_VISUAL)
-	return TRUE
+/obj/structure/loom/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/stack/sheet))
+		var/obj/item/stack/sheet/W = I
+		if(W.is_fabric && W.amount > 3)
+			user.show_message("<span class='notice'>You start weaving the [W.name] through the loom..</span>", 1)
+			if(W.use_tool(src, user, W.pull_effort))
+				new W.loom_result(drop_location())
+				user.show_message("<span class='notice'>You weave the [W.name] into a workable fabric.</span>", 1)
+				W.amount = (W.amount - FABRIC_PER_SHEET)
+				if(W.amount < 1)
+					qdel(W)
```

Comment:
```suggestion
				W.use(FABRIC_PER_SHEET)
```
also why does this use FABRIC_PER_SHEET instead of 3 when the code above checks for amount over 3?